### PR TITLE
Feature/paging-bar

### PR DESCRIPTION
**Pull request by emoralesb05**
## Description

Created standalone paging-bar component to be flexible for any use case following the specs in https://material.google.com/components/data-tables.html#data-tables-specs
https://github.com/Teradata/covalent/issues/114

#### `<td-paging-bar>`

Generate a paging bar.
The (change) event throws an event with the following interface:
```typescript
export interface IPageChangeEvent {
  page: number;
  maxPage: number;
  pageSize: number;
  total: number;
  fromRow: number;
  toRow: number;
}
```
The `<td-paging-bar>` component has 5 properties:
image

Usage in HTML:
```html
<td-paging-bar [pageSizes]="[100,200,500,1000,2000]" [pageSize]="100" [total]="1345" (change)="change($event)">
</td-paging-bar>
```

#### Test Steps

- [x] `ng serve`
- [x] Go to http://localhost:4200/#/components/paging-bar
- [x] Play with demo.

##### Screenshots or link to CodePen/Plunker/JSfiddle
image
td-paging-bar-demo

### CODE REVIEW

**Comment by kyleledbetter** [comment]
nice! should we allow a selection of "All" pages in the menu?
image

**Comment by kyleledbetter** [comment]
Optimized SCSS and split out theme:

image

**Comment by kyleledbetter** [comment]
## Updates 

- [ ] make first & last page buttons optional
- [ ] add "All" in pageSizeMenu if possible

File: src/platform/paging/paging-bar.component.ts [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+  private _toRow: number = 1;
+  private _initialized: boolean = false;
+
+  /**
+   * pageSizes?: number[]
+   * Array that populates page size menu. Defaults to [50, 100, 200, 500, 1000]
+   */
+  @Input('pageSizes')
+  set pageSizes(pageSizes: number[]) {
+    if (pageSizes instanceof Array) {
```
Comment (emoralesb05) [inline src/platform/paging/paging-bar.component.ts]: Yeah, will add an error message if its not a valid input :+1:

**Comment by emoralesb05** [comment]
I think `all` rows defeats the purpose of a paging component. 
The consumers can just set a really big number if they realllyyy wanna do that.

**Comment by kyleledbetter** [comment]
The code would have to be changed to do that, while the end user would be the one to select All. It's a ux thing if you're not wanting to go through all pages to view everything and very nice to have

**Comment by emoralesb05** [comment]
I guess what im trying to say is, if they wanna see all rows they could either turn off paging or add the total as pageSize option.
e.g.
`[pageSizes]="[100,200,500,1000, data.length]"`

**Comment by kyleledbetter** [comment]
`[pageSizes]="[100,200,500,1000, data.length]"` still puts it in the developers hands, not the end user. The end user would often want to just display All

**Comment by emoralesb05** [comment]
Right, but its up to the developer to do something with that `all` either way, so the developer still need to do it.

**Comment by emoralesb05** [comment]
Alright, you convinced me. Optional, default to false. lol

**Comment by emoralesb05** [comment]
- [x] added `pageSizeAll` input
```
<td-paging-bar [firstLast]="false" [pageSizeAll]="true" [pageSizes]="[100,200,500,1000,2000]" [pageSize]="100" [total]="1345" (change)="change($event)">
</td-paging-bar>
```
Output:
image

**Pull request merged**